src/Common/src/System/Globalization/IdnMapping.cs

```diff
@@ -80,20 +80,23 @@ public string GetAscii(string unicode, int index, int count)
             if (index > unicode.Length - count)
                 throw new ArgumentOutOfRangeException(nameof(unicode), SR.ArgumentOutOfRange_IndexCountBuffer);
             Contract.EndContractBlock();
-
-            // We're only using part of the string
-            unicode = unicode.Substring(index, count);
-
-            if (unicode.Length == 0)
+            
+            if (count == 0)
             {
                 throw new ArgumentException(SR.Argument_IdnBadLabelSize, nameof(unicode));
             }
-            if (unicode[unicode.Length - 1] == 0)
+            if (unicode[index + count - 1] == 0)
             {
-                throw new ArgumentException(SR.Format(SR.Argument_InvalidCharSequence, unicode.Length - 1), nameof(unicode));
+                throw new ArgumentException(SR.Format(SR.Argument_InvalidCharSequence, index + count - 1), nameof(unicode));
```

Comment:
I think this should be just `count`?  In the old code we used `unicode.Length` which happened after the substring, so `unicode` should be of `count` length?